src/components/Modal/context.js

```diff
@@ -0,0 +1,10 @@
+import React from "react";
+
+const { Provider, Consumer } = React.createContext();
+
+export { Provider as ModalProvider };
+export { Consumer as ModalConsumer };
+
+export const withModal = Component => props => (
+  <Consumer>{value => <Component modal={value} {...props} />}</Consumer>
```

Comment:
Why is this necessary?

Reply:
This one would help to avoid cluttering `render` code with endless callbacks.

I expect that users will use it like this:

```
const MyComponent = ({ closeModal }) => (
  <Modal>
    <div onClick={closeModal}>
      My Simple Modal
    </div>
  </Modal>
);

export default withModal(MyComponent);
```